src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -118,8 +117,7 @@ public virtual void Init(IProviderConfiguration providerCfg, string providerName
             serviceProvider = svcProvider;
             receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
             this.SerializationManager = this.serviceProvider.GetRequiredService<SerializationManager>();
-            this.getNodeConfig = svcProvider.GetRequiredService<Func<NodeConfiguration>>();
-
+            this.getNodeConfig = new Lazy<Func<NodeConfiguration>>(() => this.serviceProvider.GetRequiredService<Func<NodeConfiguration>>());
```

Comment:
yes I can do that too! great advice. we must be smoking something bad when we did this